src/driver/safe/device_ptr.rs

```diff
@@ -1,8 +1,9 @@
 use crate::driver::sys;
 
-use super::core::{CudaSlice, CudaView, CudaViewMut};
+use super::core::{CudaSlice, CudaStream, CudaView, CudaViewMut};
 
 pub trait DeviceSlice<T> {
+    fn stream(&self) -> &CudaStream;
```

Comment:
This is a breaking change. We really only need this if we are tracking what stream a CudaSlice/CudaView/CudaViewMut is owned by